### DISCUSSION TITLE
Re-enabling Corosync/Cluster/HA after Proxmox VE Post Install disabled them?

### DISCUSSION THREAD

**Discussion by BlackrazorNZ** [tteck/Proxmox#1527] [Help Needed] [answered]
Hi there,

How do I re-enable the Cluster parts of Proxmox after I previously had the Proxmox VE Post Install script disable them on initial install?

I only had 1 machine, however now I've added a 2nd and want to cluster them.

corosync, pve-ha-crm and pve-ha-lrm all show as disabled under node-System. I can start the latter 2 but not corosync, but I cannot switch their from disabled to enabled. 
![image](https://github.com/tteck/Proxmox/assets/106493161/0fdd9212-d18d-42f8-a312-7f9606c24c12)

In the new machine, all 3 services are enabled.

Addtionally, in the original machine, no CIDR shows under the Create Cluster function:
![image](https://github.com/tteck/Proxmox/assets/106493161/7149c460-84cd-4d52-b0b3-07b1e4d0ff9c)

The new 2nd machine, however, does have this:
![image](https://github.com/tteck/Proxmox/assets/106493161/4ad8868b-5c3a-424e-a351-8fbac8d57f96)

I have tried re-running the original script as I note many others have a 'second use' different workflow or toggle option, however the option to reverse the original change is not presented if you re-run the script.

Any advice on how to manually enable these options?

Thank you!

**Comment by tteck** [comment, marked as answer]
```
systemctl enable --now pve-ha-lrm
systemctl enable --now pve-ha-crm
systemctl enable --now corosync
```

**Reply by BlackrazorNZ** [reply]
Thanks - worked well. Any idea what to enable to get the network address showing up as a CIDR in Create Cluster?

 I rebooted post running the above and the services are enabled but that field is still blank. 

Given it’s pre-populated on the second machine I assume its omission is related to the above.

**Reply by tteck** [reply]
https://pve.proxmox.com/wiki/Cluster_Manager

**Reply by BlackrazorNZ** [reply]
No, it's not possible to enter it manually, you have to select an option from the drop down menu.

However I figured out what was causing it, so I thought I'd comment in case anyone else in in this circumstance. I have my Proxmox instance set to DHCP with a fixed IP address allocated by my DHCP server - I do this primarily for visibility purposes, and it works fine to date.

However, apparently unless you have an IP address physically set in /etc/network/interfaces, it will not show up in the Create Cluster CIDR dropdown. Here's my DHCP setup below - if I comment out the DHCP line and uncomment the static IP lines, the CIDR item shows up in the dropdown on reboot and then works fine.

I am going to do some more experimenting to identify if it's possible to manually set up the cluster via CLI while still having DHCP allocation - if so, I will update here.

![image](https://github.com/tteck/Proxmox/assets/106493161/ddc7d128-c86b-4ecc-bffc-a39e285a48ed)

**Reply by tteck** [reply]
Got it, so if you hadn't modified the `/etc/network/interfaces` file it would have worked after you re-enabled those services.

**Reply by tteck** [reply]
Why not set the static IP address and assign a DHCP reservation with the same IP?

**Reply by BlackrazorNZ** [reply]
Yes, that is what I will likely do. Thank you for your help :)
